summary/summary.go

```diff
@@ -68,11 +68,21 @@ func (s *StatsSummary) AddSample(stat info.ContainerStats) error {
 	s.secondSamples = append(s.secondSamples, &sample)
 	s.updateLatestUsage()
 	// TODO(jnagal): Use 'available' to avoid unnecessary computation.
-	if len(s.secondSamples) == 60 {
-		// Make a minute stat.
+	numSamples := len(s.secondSamples)
+	elapsed := int64(0)
+	if numSamples > 1 {
+		start := s.secondSamples[0].Timestamp
+		end := s.secondSamples[numSamples-1].Timestamp
+		elapsed = end.Sub(start).Nanoseconds()
+	}
+	if elapsed > 60*time.Second.Nanoseconds() {
```

Comment:
yeah, we add one stat at a time and also max dynamic housekeeping is currently limited to 1minute (with a flag). The only way this can go way beyond 60s is if we  increase the max allowed housekeeping time. Testing on empty containers generate 7 and 2 samples every other minute (due to backoff).